### ISSUE TITLE
To OOP or not to OOP?

### ISSUE THREAD

**Issue by clstaudt** [tuttle-dev/tuttle#133]
Pick the better implementation:

A)

```python
def get_dropdown(
    lbl: str,
    on_change: Callable,
    items: List[str],
    hint: Optional[str] = "",
    width: Optional[int] = None,
    initial_value: Optional[str] = None,
):
    options = []
    for item in items:
        options.append(
            dropdown.Option(
                text=item,
            )
        )
    return Dropdown(
        label=lbl,
        hint_text=hint,
        options=options,
        text_size=fonts.BODY_1_SIZE,
        label_style=TextStyle(size=fonts.BODY_2_SIZE),
        on_change=on_change,
        width=width,
        value=initial_value,
        content_padding=padding.all(dimens.SPACE_XS),
        error_style=TextStyle(size=fonts.BODY_2_SIZE, color=colors.ERROR_COLOR),
    )
```

B) 

```python
class StandardDropdown(Dropdown):
    def __init__(
        self,
        lbl: str,
        on_change: Callable,
        items: List[Union[str, dropdown.Option]],
        hint: Optional[str] = "",
        width: Optional[int] = None,
        initial_value: Optional[str] = None,
    ):
        options = []
        for item in items:
            options.append(
                dropdown.Option(
                    text=item,
                )
            )


        super().__init__(
            label=lbl,
            hint_text=hint,
            options=options,
            text_size=fonts.BODY_1_SIZE,
            label_style=TextStyle(size=fonts.BODY_2_SIZE),
            on_change=on_change,
            width=width,
            value=initial_value,
            content_padding=padding.all(dimens.SPACE_XS),
            error_style=TextStyle(size=fonts.BODY_2_SIZE, color=colors.ERROR_COLOR),
        )

```

**Comment by vlad-ed-git** [comment]
@clstaudt for these views, A is better. because we want to be able to just do `views.get_dropdown()` in our Ui without having to instantiate a class object every time you set a view.

**Comment by clstaudt** [comment]
Can't follow that explanation, can you rephrase?

**Comment by vlad-ed-git** [comment]
If we make the view a class, suppose it's a drop-down.
Then to add it to a column, we have to do
1. Instantiate the class
2. Pass the object to the column
   e.g. drop_down = views.TuttleDropdown()
   column.controls = [drop_down]

The current option is just
   column.controls = [views.getDropdown()]

**Comment by clstaudt** [comment]
```python
column.controls = [views.TuttleDropdown()]
```
